src/overlays/actors/ovl_Bg_F40_Switch/z_bg_f40_switch.c

```diff
@@ -34,37 +34,167 @@ const ActorInit Bg_F40_Switch_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Switch/BgF40Switch_Draw.s")
+void BgF40Switch_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgF40Switch* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor.actor, sInitChain);
+    this->actor.actor.scale.y = 0.165f;
+    this->actionFunc = BgF40Switch_ActionIdleNotPressed;
+    this->actor.actor.world.pos.y = this->actor.actor.home.pos.y + 1.0f;
+    DynaPolyActor_Init(&this->actor, 1);
+    DynaPolyActor_LoadMesh(globalCtx, &this->actor, &object_f40_switch_Colheader_000118);
+    if (!sBgF40SwitchGlobalsInitialized) {
+        sBgF40SwitchLastUpdateFrame = globalCtx->gameplayFrames;
+        sBgF40SwitchGlobalsInitialized = true;
+    }
+}
+
+void BgF40Switch_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgF40Switch* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->actor.bgId);
+}
+
+void BgF40Switch_ActionBecomeNotPressed(BgF40Switch* this, GlobalContext* globalCtx) {
+    this->actor.actor.scale.y += 0.0495f;
+    if (this->actor.actor.scale.y >= 0.165f) {
+        Audio_PlayActorSound2(&this->actor.actor, NA_SE_EV_IKANA_BLOCK_SWITCH);
+        this->actionFunc = BgF40Switch_ActionIdleNotPressed;
+        this->actor.actor.scale.y = 0.165f;
+    }
+}
+
+void BgF40Switch_ActionIdlePressed(BgF40Switch* this, GlobalContext* globalCtx) {
+    if (!this->isPressed) {
+        this->actionFunc = BgF40Switch_ActionBecomeNotPressed;
+    }
+}
+
+void BgF40Switch_ActionBecomePressed(BgF40Switch* this, GlobalContext* globalCtx) {
+    this->actor.actor.scale.y -= 0.0495f;
+    if (this->actor.actor.scale.y <= 0.0165f) {
+        Audio_PlayActorSound2(&this->actor.actor, NA_SE_EV_IKANA_BLOCK_SWITCH);
+        func_8013ECE0(this->actor.actor.xyzDistToPlayerSq, 120, 20, 10);
+        if (this->isInitiator) {
+            ActorCutscene_Stop(this->actor.actor.cutscene);
+            this->isInitiator = false;
+        }
+        this->actionFunc = BgF40Switch_ActionIdlePressed;
+        this->actor.actor.scale.y = 0.0165f;
+        this->switchReleaseDelay = 6;
+    }
+}
+
+void BgF40Switch_ActionWaitToBecomePressed(BgF40Switch* this, GlobalContext* globalCtx) {
+    if (this->isInitiator == 0 || this->actor.actor.cutscene == -1) {
+        this->actionFunc = BgF40Switch_ActionBecomePressed;
+        if (this->isInitiator) {
+            Actor_SetSwitchFlag(globalCtx, GET_BGF40SWITCH_SWITCHFLAG(this));
+        }
+    } else {
+        if (ActorCutscene_GetCanPlayNext(this->actor.actor.cutscene)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->actor.actor.cutscene, &this->actor.actor);
+            this->actionFunc = BgF40Switch_ActionBecomePressed;
+            if (this->isInitiator) {
+                Actor_SetSwitchFlag(globalCtx, GET_BGF40SWITCH_SWITCHFLAG(this));
+            }
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.actor.cutscene);
+        }
+    }
```

Comment:
You can merge that into a single `else if`
```suggestion
        }
    } else if (ActorCutscene_GetCanPlayNext(this->actor.actor.cutscene)) {
        ActorCutscene_StartAndSetUnkLinkFields(this->actor.actor.cutscene, &this->actor.actor);
        this->actionFunc = BgF40Switch_ActionBecomePressed;
        if (this->isInitiator) {
            Actor_SetSwitchFlag(globalCtx, GET_BGF40SWITCH_SWITCHFLAG(this));
        }
    } else {
        ActorCutscene_SetIntentToPlay(this->actor.actor.cutscene);
    }
```